tests/integration/targets/pacman/tasks/package_urls.yml

```
@@ -15,11 +19,31 @@
           - '{{file_pkg}}'
           - '{{extra_pkg}}'
         state: absent
+    - name: Make sure that url package is not cached
+      file:
+        path: '/var/cache/pacman/pkg/{{url_pkg_filename}}'
+        state: absent
 
     - name: Get URL for {{url_pkg}}
       command:
         cmd: pacman --sync --print-format "%l" {{url_pkg}}
-      register: url_pkg_url
+      register: url_pkg_stdout
+    - name: Download {{url_pkg}} pkg
+      get_url:
+        url: '{{url_pkg_stdout.stdout}}'
+        dest: '{{url_pkg_path}}/{{url_pkg_filename}}'
+    - name: Download {{url_pkg}} pkg sig
+      get_url:
+        url: '{{url_pkg_stdout.stdout}}.sig'
+        dest: '{{url_pkg_path}}/{{url_pkg_filename}}.sig'
+    - name: Host {{url_pkg}}
+      shell:
+        cmd: 'python -m http.server --directory {{url_pkg_path}} {{http_port}} >/dev/null 2>&1 &'
+    - name: Wait for http.server to come up online
+      wait_for:
+        host: 'localhost'
+        port: '{{http_port}}'
+        state: started
```

I would wrap this in a `block:` and add a `always:` part which makes sure to kill that server.
```suggestion
    - block:
      - name: Host {{url_pkg}}
        shell:
          cmd: 'python -m http.server --directory {{url_pkg_path}} {{http_port}} >/dev/null 2>&1 &'
      - name: Wait for http.server to come up online
        wait_for:
          host: 'localhost'
          port: '{{http_port}}'
          state: started
```

---

I now copied the solution from the integration tests of `get_url`, as this uses ansibles builtin asynchronous tasks with a timeout. This should be a much cleaner solution for asynchronous tasks then the previously used ampersand. On the other hand, this results in the server always running 90 seconds, even if the play runs longer or shorter. But I currently have no good idea on how to properly kill the server without killing the used python module.

---

I think that's ok.